server.go

```diff
@@ -915,11 +915,19 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB,
 				return ErrServerShuttingDown
 			}
 		},
-		DisableChannel:      s.chanStatusMgr.RequestDisable,
-		Sweeper:             s.sweeper,
-		Registry:            s.invoices,
-		NotifyClosedChannel: s.channelNotifier.NotifyClosedChannelEvent,
-		OnionProcessor:      s.sphinx,
+		DisableChannel:               s.chanStatusMgr.RequestDisable,
+		Sweeper:                      s.sweeper,
+		Registry:                     s.invoices,
+		NotifyClosedChannel:          s.channelNotifier.NotifyClosedChannelEvent,
+		OnionProcessor:               s.sphinx,
+		PaymentsExpirationGracePriod: cfg.PaymentsExpirationGracePriod,
+		IsInitiatedPayment: func(hash [32]byte) (bool, error) {
+			payment, err := paymentControl.FetchPayment(hash)
```

Comment:
This looks like a layering violation. Payments are a higher level concept and can get complicated with mpp (multiple outgoing htlcs with the same hash) and amp (multiple htlcs with different hashes). Maybe it is better to check the circuit map to see if this is a forward?

Reply:
I agree. I think instead we can query the Switch for whether this is a forward.
@joostjager  Is it correct to say that if the field `AddRef` of `PaymentCircuit` struct is **NOT** nil then we are dealing with a forward?

Reply:
I think we mostly check `incomingChanID == hop.Source`

Reply:
@joostjager  The implementation of the check for forward/initiated htlc is now moved to the switch.